components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/dao/impl/handler/PublisherAdapterTypeHandler.java

```diff
@@ -169,7 +170,28 @@ public void updateWebhook(Webhook webhook, int tenantId) throws WebhookMgtExcept
     @Override
     public List<Webhook> getActiveWebhooks(String eventProfileName, String eventProfileVersion, String channelUri,
                                            int tenantId) throws WebhookMgtException {
-
-        return dao.getActiveWebhooks(eventProfileName, eventProfileVersion, channelUri, tenantId);
+        // Get active webhooks from DAO.
+        List<Webhook> webhooks = dao.getActiveWebhooks(eventProfileName, eventProfileVersion, channelUri, tenantId);
+
+        // Create a new list with decrypted secrets.
+        List<Webhook> decryptedWebhooks = new ArrayList<>();
+        for (Webhook webhook : webhooks) {
+            String decryptedSecret = getWebhookDecryptedSecretValue(webhook.getId());
+            Webhook decryptedWebhook = new Webhook.Builder()
+                    .uuid(webhook.getId())
+                    .endpoint(webhook.getEndpoint())
+                    .name(webhook.getName())
+                    .secret(decryptedSecret)
+                    .eventProfileName(webhook.getEventProfileName())
+                    .eventProfileUri(webhook.getEventProfileUri())
+                    .eventProfileVersion(webhook.getEventProfileVersion())
+                    .status(webhook.getStatus())
+                    .createdAt(webhook.getCreatedAt())
+                    .updatedAt(webhook.getUpdatedAt())
+                    .eventsSubscribed(webhook.getEventsSubscribed())
+                    .build();
+            decryptedWebhooks.add(decryptedWebhook);
+        }
+        return decryptedWebhooks;
```

Comment:
Since we are caching this list, is it okay to keep the webhooks with decrypted secrets in the memory ?

Reply:
Fixed via https://github.com/wso2/carbon-identity-framework/pull/7323/commits/f036fb569e8961d0f0265507ce73b576a95ae134